backend/routes/admin_refresh.py

```
@@ -0,0 +1,98 @@
+        logger.info(f"Refresh flag set: force_reindex={request.force_reindex}")
+
+        return {
+            "message": "Refresh flag set successfully",
+            "force_reindex": request.force_reindex,
+            "timestamp": datetime.now().isoformat(),
+            "note": "Changes will take effect on next server restart",
+        }
+
+    except Exception as e:
+        logger.error(f"Failed to set refresh flag: {e}", exc_info=True)
+        raise HTTPException(status_code=500, detail=f"Failed to set refresh flag: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)